### PR TITLE
feat: support tokens scoped to multiple repositories within organization

### LINKED ISSUE #4: Support tokens with access to custom repositories and user/organization accounts
The use cases we want to cover is

1. By default, create a token scoped to the current repository
2. Allow to scope to a set of repositories by names for the current owner
3. Allow to scope to a different owner (all installation repositories)
4. Allow to scope to a set of repositories by names for a different owner

Our API idea is

```yml
      - uses: actions/create-github-app-token@v1
        id: app-token
        with:
          app_id: ${{ vars.APP_ID }}
          private_key: ${{ secrets.PRIVATE_KEY }}
          owner: some-owner
          repositories: some-repo1,some-repo2
```

Both `owner` and `repositories` would be optional, and one can be set to the other. `owner` defaults to the current repository owner, and repositories defaults to the current repository name.

### PR DESCRIPTION
**Pull request by timreimherr**
This adds support for a token to be scoped to multiple repositories within an organization.

resolves #4

### CODE REVIEW

**Review by gr2m — CHANGES_REQUESTED** [review]
Thank you for the pull request!

I think there are two use cases we should support which are not covered by your code changes

1. Create token for current organization but for all repositories
2. Create token for a given organization for all repositories

WIth these use cases things will get a bit complicated and I think we should add tests for them. I can help with the tests if needed

**Comment by timreimherr** [comment]
@gr2m thanks for the feedback! Just pushed the changes you requested. Started to work on tests but my lack of time kept me from tests. So, I'll definitely take your offer to help with the tests! 🙇 👍

**Review by gr2m — COMMENTED** [review]
did you try if it works as expected? You can update the build file and then use `uses: timreimherr/create-github-app-token`. I think if you don't set `owner` and `repositories`, then it will now create a token that has access to all repositories of the current repository owner's installation, instead of defaulting to just the current repository only

File: README.md
Comment (gr2m) [inline README.md:12]: I don't think I would recommend this in general. I would recommend it when describing the `owner` and `repositories` arguments. We can add another usage example for a token that has access to all repositories in the current repository owner's installation

File: README.md
Comment (gr2m) [inline README.md:14]: We can rename it to make clear what it dose

```suggestion
### Create token for current repository
```

I would then add more usage examples

1. Create token for all repositories in current owner's installation
2. Create a token for selected repositories in current owner's installation
3. Create token for all repositories in another owner's installation

Note that an owner can be both an organization or a user

File: README.md
Comment (gr2m) [inline README.md:29]: remove from this usage example, see above

File: README.md
Comment (gr2m) [inline README.md:55]: remove from this usage example, I'd focus on the use case

File: README.md
Comment (gr2m) [inline README.md:83]:
```suggestion
**Optional:** Comma-separated list of repositories to grant access to. If 'owner' is set and 'repositories' is empty then the access will be scoped to all repositories in the current repository owner's installation. If 'owner' and 'repositories' are empty then the access will be scoped to the current repository.
```

File: README.md
Comment (timreimherr) [reply README.md]: Ah gotcha, that makes sense. Done!

File: README.md
Comment (timreimherr) [reply README.md]: Removed in favor of use cases.

File: README.md
Comment (timreimherr) [reply README.md]: Done.

File: README.md
Comment (timreimherr) [reply README.md]: Done.

File: README.md
Comment (timreimherr) [reply README.md]: Done.

File: action.yml
Comment (qoomon) [inline action.yml:1]: Why it is `Tim's` ?

Comment (gr2m) [reply action.yml:1]: I think Time wanted to publish the action to the marketplace in order to run the action from the fork, but publishing to the marketplace is not necessary. We definitely need to revert this change before merging

Comment (timreimherr) [reply action.yml:1]: Yes, that is exactly why. I'll revert it, post haste.

**Comment by collinmcneese** [comment]
This is wonderful and extends the functionality of this action by quite a bit! 🎉

File: lib/main.js
Comment (gr2m) [inline lib/main.js:49]: if we need to get the installation ID for the current owner, we should keep doing it using `GET /repos/{owner}/{repo}/installation`. Because if the owner is set to something else, we need to check two endpoints, because the owner can be either an org or a user:

1. [GET /orgs/{org}/installation](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-a-repository-installation-for-the-authenticated-app)
2. [GET /users/{username}/installation](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-a-user-installation-for-the-authenticated-app)

**Review by gr2m — COMMENTED** [review]
I also updated the `dist/main.cjs` file

File: lib/main.js
Comment (gr2m) [reply lib/main.js:49]: See my suggested change at https://github.com/timreimherr/create-github-app-token/pull/1

File: dist/main.cjs
Comment (tjorvi) [inline dist/main.cjs:15069]: `org` is always empty here if owner2 is passed to this function

Comment (gr2m) [reply dist/main.cjs:15069]: The `dist/*` files are compiled build files. Better comment on the `main.js` and `lib/main.js`

I think that my PR here addresses your concern https://github.com/timreimherr/create-github-app-token/pull/1

Comment (tjorvi) [reply dist/main.cjs:15069]: This was an issue I ran into when trying to use your fork as an action. It might be that the dist file wasn't up to date at the time I made the comment, I just assumed it would be.

Comment (gr2m) [reply dist/main.cjs:15069]: ah makes sense, thank you for clarifying

File: README.md
Comment (gr2m) [inline README.md:143]:
```suggestion
**Optional:** Comma-separated list of repositories to grant access to. If 'owner' is set and 'repositories' is empty then the access will be scoped to all repositories in the provided repository owner's installation. If 'owner' and 'repositories' are empty then the access will be scoped to only the current repository.
```

File: action.yml
Comment (gr2m) [inline action.yml:18]:
```suggestion
    description: "Repositories to install the GitHub App on (defaults to current repository if owner is unset)"
```

**Review by gr2m — APPROVED** [review]
still lacking tests but we can merge it as is now, I tested it manually. Waiting for review by @parkerbxyz to make sure I didn't miss anything

File: lib/main.js
Comment (parkerbxyz) [inline lib/main.js:123]:
```suggestion
  // Register the token with the runner as a secret to ensure it is masked in logs
  core.setSecret(authentication.token);

  core.setOutput("token", authentication.token);
  
  // Make token accessible to post function (so we can invalidate it)
  core.saveState("token", authentication.token);
```

It looks like the order of `setOutput` and `setSecret` were reversed. Was this intentional?

Comment (gr2m) [reply lib/main.js:123]: nope that was me, I didn't know the order matters 😁

**Review by parkerbxyz — APPROVED** [review]

**Pull request merged**
